app/assets/src/components/views/discovery/DiscoveryView.jsx

```diff
@@ -1012,22 +1038,121 @@ class DiscoveryView extends React.Component {
     }
   };
 
+  handleEmptyStateModalClose = () => {
+    this.setState({
+      emptyStateModalOpen: false,
+    });
+  };
+
+  renderNoDataBanners = () => {
+    const { currentTab, emptyStateModalOpen, userDataCounts } = this.state;
+    const { visualizations } = this;
+
+    if (!userDataCounts) return null;
+
+    if (emptyStateModalOpen) {
+      localStorage.setItem("DiscoveryViewSeenBefore", "1");
+    }
+
+    switch (currentTab) {
+      case "projects":
+        if (userDataCounts.projectCount === 0) {
+          return emptyStateModalOpen ? (
+            <EmptyStateModal onClose={this.handleEmptyStateModalClose} />
+          ) : (
+            <div className={cs.noDataBannerFlexContainer}>
+              <InfoBanner
+                className={cs.noDataBannerContainer}
+                icon={<BannerProjects />}
+                link={{
+                  href: "/samples/upload",
+                  text: "Upload your data",
+                }}
+                message="You will see your projects here after you upload data or when you are invited to a project."
+                title="Projects"
+                type="no_projects"
+              />
+            </div>
+          );
+        }
+        break;
+      case "samples":
+        if (userDataCounts.sampleCount === 0) {
+          return emptyStateModalOpen ? (
+            <EmptyStateModal onClose={this.handleCloseEmptyStateModal} />
+          ) : (
+            <div className={cs.noDataBannerFlexContainer}>
+              <InfoBanner
+                className={cs.noDataBannerContainer}
+                icon={<BannerSamples />}
+                link={{
+                  href: "/samples/upload",
+                  text: "Upload your data",
+                }}
+                message="You will see your samples here after you upload data or when you are invited to a project."
+                title="Samples"
+                type="no_samples"
+              />
+            </div>
+          );
+        }
+        break;
+      case "visualizations":
+        // visualizations are not filtered by conditions (not supported at this time)
+        // thus we use the object collection view from DataDiscoveryDataLayer directly
+        // and we never show the no search results banner.
+        if (!visualizations.isLoading() && visualizations.length === 0) {
+          return emptyStateModalOpen ? (
+            <EmptyStateModal onClose={this.handleCloseEmptyStateModal} />
+          ) : (
+            <div className={cs.noDataBannerFlexContainer}>
+              <InfoBanner
+                className={cs.noDataBannerContainer}
+                icon={<BannerVisualizations />}
+                link={{
+                  href: VISUALIZATIONS_DOC_LINK,
```

Comment:
typically we make links to the help center open in a new tab but not a big deal

Reply:
Good point. Added option for external link.